Assets/Scripts/Managers/BuddyManager.cs

```diff
@@ -14,6 +17,17 @@ void Start()
 	public static void RegisterBuddy( BuddyStats buddyStats )
 	{
 		instance._buddyStats.Add( buddyStats );
+
+		instance.ActivateCheckpoint();
+	}
+
+	void ActivateCheckpoint()
+	{
+		if( checkpointIndex < checkpointLifters.Length && checkpointLifters[checkpointIndex] != null )
```

Comment:
I think the condition after the && here could just be "checkpointLifters[checkpointIndex]"